### ISSUE TITLE
New SEACAS tests failing in ATDM Trilinos builds starting on 7/19/2018 and 7/23/2018

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#3183]
New SEACAS tests failing in ATDM Trilinos builds starting on 7/19/2018 and 7/23/2018

CC: @trilinos/seacas, @gsjaardema (pushed breaking commits?), @kddevin (Trilinos Data Services Product Lead)

## Next Action Status

PR #3213 merged on 8/1/2018 then later fixed in PR #3251 merged 8/8/2018 that disabled most of these tests in the 'mutrino' builds on 8/2/2018.  Next: Disable randomly failing test `SEACASAprepro_aprepro_test_dump_reread` ...

## Description

As shown in [this query for the builds today](https://testing-vm.sandia.gov/cdash/queryTests.php?project=Trilinos&date=2018-07-24&filtercombine=and&filtercombine=and&filtercount=5&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=status&compare2=62&value2=passed&field3=status&compare3=62&value3=notrun&field4=groupname&compare4=61&value4=ATDM&field5=testname&compare5=65&value5=SEACAS), the tests:

* SEACASAprepro_aprepro_array_test
* SEACASAprepro_aprepro_command_line_include_test
* SEACASAprepro_aprepro_command_line_vars_test
* SEACASAprepro_aprepro_unit_test
* SEACASAprepro_lib_aprepro_lib_array_test
* SEACASAprepro_lib_aprepro_lib_unit_test
* SEACASExodus_exodus_unit_tests_nc5_env

are failing in the builds:

* Trilinos-atdm-mutrino-intel-debug-openmp
* Trilinos-atdm-mutrino-intel-opt-openmp

and the tests:

* SEACASIoss_exodus32_to_exodus32
* SEACASIoss_exodus32_to_exodus32_pnetcdf
* SEACASIoss_exodus32_to_exodus64

are failing in the builds:

* Trilinos-atdm-hansen-shiller-cuda-8.0-debug
* Trilinos-atdm-hansen-shiller-cuda-8.0-opt

As shown in [this query showing failing SEACAS tests going back to 7/10/2018](https://testing-vm.sandia.gov/cdash/queryTests.php?project=Trilinos&date=2018-07-24&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=6&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=status&compare2=62&value2=passed&field3=status&compare3=62&value3=notrun&field4=groupname&compare4=61&value4=ATDM&field5=testname&compare5=65&value5=SEACAS&field6=buildstarttime&compare6=83&value6=2018-07-10), the test `	SEACASExodus_exodus_unit_tests_nc5_env`	started failing on 7/19/2018 and the other tests started failing on 7/23/2018.  There were several PRs merged the days before these dates by @gsjaardema so it is not clear which changes caused these new failures but it seems likely that one or more of the commits in these merged PRs triggered these new failures.

Also, the test `SEACASAprepro_aprepro_test_dump_reread` added in one of these PRs appeared on 7/23/2018 and then started randomly failing as shown in [this query](https://testing-vm.sandia.gov/cdash/queryTests.php?project=Trilinos&date=2018-08-02&filtercombine=and&filtercombine=and&filtercount=5&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=groupname&compare2=61&value2=ATDM&field3=testname&compare3=61&value3=SEACASAprepro_aprepro_test_dump_reread&field4=site&compare4=61&value4=mutrino&field5=buildstarttime&compare5=83&value5=2018-07-15).  When the test passes like shown [here](https://testing-vm.sandia.gov/cdash/testDetails.php?test=51055647&build=3759381), it shows:

```
================================================================================

TEST_3

Running: "diff" "-w" "test-filter.dump" "test-reread.dump"

--------------------------------------------------------------------------------


--------------------------------------------------------------------------------

TEST_3: Return code = 0
TEST_3: Pass criteria = Zero return code [PASSED]
TEST_3: Result = PASSED

================================================================================
```

when it fails like shown [here](https://testing-vm.sandia.gov/cdash/testDetails.php?test=51563099&build=3790512), it shows:

```
================================================================================

TEST_3

Running: "diff" "-w" "test-filter.dump" "test-reread.dump"

--------------------------------------------------------------------------------

1,2c1,2
< Thu Aug  2 11:20:16 2018: [unset]:_pmi_alps_init:alps_get_placement_info returned with error -1
< Thu Aug  2 11:20:16 2018: [unset]:_pmi_init:_pmi_alps_init returned -1
---
> Thu Aug  2 11:20:17 2018: [unset]:_pmi_alps_init:alps_get_placement_info returned with error -1
> Thu Aug  2 11:20:17 2018: [unset]:_pmi_init:_pmi_alps_init returned -1

--------------------------------------------------------------------------------

TEST_3: Return code = 1
TEST_3: Pass criteria = Zero return code [FAILED]
TEST_3: Result = FAILED

================================================================================
```

## Steps to reproduce

These failures should be reproducable on the machines 'hansen' or 'shiller' and 'mutrino' using the instructions in:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md

For example, for the failures on 'hansen'/'shiler', the specific instructions are given at:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#shillerhansen

For example, after cloning Trilinos, the following commands should reproduce the test failures on 'hansen' or 'shiller' with:

```
$ cd <some_build_dir>/

$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh cuda-8.0-debug

$ cmake \
  -GNinja \
  -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnvSettings.cmake \
  -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_SEACAS=ON \
  $TRILINOS_DIR

$ make NP=16

$ srun ctest -j16
```

**Comment by gsjaardema** [comment]
@bartlettroscoe 
The aprepro test failures on mutrino all seem to be due to the system writing some extra data to stderr.  The test is diffing stderr with the gold stderr output.  I guess I need a better way of getting just the aprepro stderr output...  The extra data being written to stderr is:
```
0a1,2
> Tue Jul 24 08:34:53 2018: [unset]:_pmi_alps_init:alps_get_placement_info returned with error -1
> Tue Jul 24 08:34:53 2018: [unset]:_pmi_init:_pmi_alps_init returned -1
``` 
The new capability in aprepro is the addition of exodus query capability, so the extra data may be related to linking aprepro now to a parallel hdf5 and netcdf library...

I will take a look.

**Comment by gsjaardema** [comment]
@bartlettroscoe 
The `SEACASIoss` tests on hansen should be fixed with the next test run since #3167 addresses exactly the problem causing these to fail.

**Comment by bartlettroscoe** [comment]
@gsjaardema, if you don't want to deal with the hassle of fixing those tests on 'mutrino', we can just disable them on 'mutrino'.  I can provide detailed instructions on how to do that.

**Comment by gsjaardema** [comment]
@bartlettroscoe  Yes, that would be good to do.

**Comment by gsjaardema** [comment]
@bartlettroscoe How do I disable the tests on `mutrino`?

**Comment by bartlettroscoe** [comment]
> @bartlettroscoe How do I disable the tests on mutrino?

@gsjaardema, if you have some time, can you please take a look at the documentation that explains how to do this that I wrote as part of the PR #3211?  Specifically, can you read over the new section shown in this PR branch at:

* https://github.com/bartlettroscoe/Trilinos/blob/3183-disable-tests-info/cmake/std/atdm/README.md#disabling-failing-tests

and then comment in that PR if you see any problems or if anything is not clear?  Of you don't find anything wrong, can you please approve that PR so that we can merge it?

Then hopefully it will be straightforward how to disable these specific tests for the builds on 'mutrino' (which you can do in a new PR).

Note the sub-process [Temporarily disable the failing code or test ](https://snl-wiki.sandia.gov/display/CoodinatedDevOpsATDM/Triaging+and+addressing+ATDM+Trilinos+Failures#TriagingandaddressingATDMTrilinosFailures-5.Makesuretheissueisaddressedinatimelyway:).

This PR #3211 is just providing the technical details on how to disable the test.

**Comment by krcb** [comment]
Thanks!

Kris

Sent from my iPhone

On Jul 31, 2018, at 11:36 AM, Roscoe A. Bartlett <notifications@github.com<mailto:notifications@github.com>> wrote:


@bartlettroscoe<https://github.com/bartlettroscoe> How do I disable the tests on mutrino?

@gsjaardema<https://github.com/gsjaardema>, if you have some time, can you please take a look at the documentation that explains how to do this that I wrote as part of the PR #3211<https://github.com/trilinos/Trilinos/pull/3211>? Specifically, can you read over the new section shown in this PR branch at:

  *   https://github.com/bartlettroscoe/Trilinos/blob/3183-disable-tests-info/cmake/std/atdm/README.md#disabling-failing-tests

and then comment in that PR if you see any problems or if anything is not clear? Of you don't find anything wrong, can you please approve that PR so that we can merge it?

Then hopefully it will be straightforward how to disable these specific tests for the builds on 'mutrino' (which you can do in a new PR).

Note the sub-process Temporarily disable the failing code or test <https://snl-wiki.sandia.gov/display/CoodinatedDevOpsATDM/Triaging+and+addressing+ATDM+Trilinos+Failures#TriagingandaddressingATDMTrilinosFailures-5.Makesuretheissueisaddressedinatimelyway:> .

This PR #3211<https://github.com/trilinos/Trilinos/pull/3211> is just providing the technical details on how to disable the test.

**Comment by bartlettroscoe** [comment]
PR #3213 was just merged that should disable these tests on the 'mutrino' builds.  After we get confirmation that these tests are disabled tomorrow, then we can add the "Disabled Tests" labels and move one.

**Comment by bartlettroscoe** [comment]
@gsjaardema,

The good news is that the 'mutrino' build SEACAS test results today shown [here](https://testing-vm.sandia.gov/cdash/queryTests.php?project=Trilinos&date=2018-08-02&filtercombine=and&filtercombine=and&filtercount=5&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=status&compare2=62&value2=passed&field3=status&compare3=62&value3=notrun&field4=groupname&compare4=61&value4=ATDM&field5=testname&compare5=65&value5=SEACAS) don't show the following tests as failing:

* SEACASAprepro_aprepro_array_test
* SEACASAprepro_aprepro_command_line_include_test
* SEACASAprepro_aprepro_command_line_vars_test
* SEACASAprepro_aprepro_unit_test
* SEACASAprepro_lib_aprepro_lib_array_test
* SEACASAprepro_lib_aprepro_lib_unit_test
* SEACASExodus_exodus_unit_tests_nc5_env

(because they have been disabled)

The bad news is that it does show that the test `SEACASAprepro_aprepro_test_dump_reread` is failing in both of the builds.  The test `SEACASAprepro_aprepro_test_dump_reread` appeared in testing on 7/23/2018 (and therefore must have been a PR merged to 'develop' on 7/22/2018) and then started randomly failing as shown in [this query](https://testing-vm.sandia.gov/cdash/queryTests.php?project=Trilinos&date=2018-08-02&filtercombine=and&filtercombine=and&filtercount=5&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=groupname&compare2=61&value2=ATDM&field3=testname&compare3=61&value3=SEACASAprepro_aprepro_test_dump_reread&field4=site&compare4=61&value4=mutrino&field5=buildstarttime&compare5=83&value5=2018-07-15). 

When the test passes like shown [here](https://testing-vm.sandia.gov/cdash/testDetails.php?test=51055647&build=3759381), it shows:

```
================================================================================

TEST_3

Running: "diff" "-w" "test-filter.dump" "test-reread.dump"

--------------------------------------------------------------------------------


--------------------------------------------------------------------------------

TEST_3: Return code = 0
TEST_3: Pass criteria = Zero return code [PASSED]
TEST_3: Result = PASSED

================================================================================
```

when it fails like shown [here](https://testing-vm.sandia.gov/cdash/testDetails.php?test=51563099&build=3790512), it shows:

```
================================================================================

TEST_3

Running: "diff" "-w" "test-filter.dump" "test-reread.dump"

--------------------------------------------------------------------------------

1,2c1,2
< Thu Aug  2 11:20:16 2018: [unset]:_pmi_alps_init:alps_get_placement_info returned with error -1
< Thu Aug  2 11:20:16 2018: [unset]:_pmi_init:_pmi_alps_init returned -1
---
> Thu Aug  2 11:20:17 2018: [unset]:_pmi_alps_init:alps_get_placement_info returned with error -1
> Thu Aug  2 11:20:17 2018: [unset]:_pmi_init:_pmi_alps_init returned -1

--------------------------------------------------------------------------------

TEST_3: Return code = 1
TEST_3: Pass criteria = Zero return code [FAILED]
TEST_3: Result = FAILED

================================================================================
```

This looks like the same problem you mentioned [above](https://github.com/trilinos/Trilinos/issues/3183#issuecomment-407548369) with these annoying STDERR output on 'mutrino'.  The reason we did not notice this before was that these are randomly failing tests and the day that I looked at to create this issue did not happen to have these tests failing.

Can we disable this test as well in the 'mutrino' builds?  And can we refactor to a single `*.cmake` file all of the common test disabled as described [here](https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#disable-a-test-for-several-or-all-builds-on-a-specific-platform)?

**Comment by gsjaardema** [comment]
Yes.  That would be good to do.  Should I do it or do you want to.

.. gteg

**Comment by bartlettroscoe** [comment]
@gsjaardema, I can create a new PR to disable this last test after the PR #3051 gets merged.  Otherwise, these PRs could conflict (depending on how smart git/github is).

**Comment by bartlettroscoe** [comment]
These tests were shown failing again is the build  `Trilinos-atdm-mutrino-intel-opt-openmp-HSW` yesterday as shown [here](https://testing-vm.sandia.gov/cdash/queryTests.php?project=Trilinos&filtercombine=and&filtercombine=and&date=2018-08-05&filtercombine=and&filtercount=4&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=status&compare2=62&value2=passed&field3=status&compare3=62&value3=notrun&field4=groupname&compare4=61&value4=ATDM).  The disables for these tests is not showing up in the configure output for this build shown [here](https://testing-vm.sandia.gov/cdash/viewConfigure.php?buildid=3798062).  I will figure out what happened and while I am at it, I will disable the last test `SEACASAprepro_aprepro_test_dump_reread`.

**Comment by bartlettroscoe** [comment]
I can't explain what happened, but somehow the  commit a133d8642643637115b20720926428c852402322:

```
commit a133d8642643637115b20720926428c852402322
Author:     Joe Frye <jfrye@sandia.gov>
AuthorDate: Wed Jul 18 13:45:59 2018 -0600
Commit:     Roscoe A. Bartlett <rabartl@sandia.gov>
CommitDate: Thu Aug 2 09:27:48 2018 -0600

    Add KOKKOS_ARCH to ATDM_JOB_NAME_KEYS_STR and in all_supported_builds and tweaks files (#2680)
    
    This allows for different tweaks for multiple KOKKOS_ARCH values on the same
    system.
    
    Issue: #2680
    
    This changes the names of the existing builds on 'mutrino' to have an explicit
    HSW in the build name.

R100    cmake/std/atdm/mutrino/tweaks/INTEL-DEBUG-OPENMP.cmake  cmake/std/atdm/mutrino/tweaks/INTEL-DEBUG-OPENMP-HSW.cmake
R100    cmake/std/atdm/mutrino/tweaks/INTEL-RELEASE-OPENMP.cmake        cmake/std/atdm/mutrino/tweaks/INTEL-DEBUG-OPENMP-KNL.cmake
A       cmake/std/atdm/mutrino/tweaks/INTEL-RELEASE-OPENMP-HSW.cmake
A       cmake/std/atdm/mutrino/tweaks/INTEL-RELEASE-OPENMP-KNL.cmake
```

dropped the disables that @gsjaardema added in commit 5059d8a42117b55268326b7742232ab2ab9f8bd1.

This was a simple file renaming so I can't understand how it deleted these lines.  Very scary.

I will add the disables again.

**Comment by bartlettroscoe** [comment]
FYI: PR #3251 should contain the necessary disables and removed duplication.

@gsjaardema or @fryeguy52, can you please approve PR #3251 so that I can merge once PR testing is complete?

**Comment by bartlettroscoe** [comment]
FYI: As shown in [this query](https://testing-vm.sandia.gov/cdash/queryTests.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&date=2018-08-27&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=7&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=testname&compare2=61&value2=SEACASAprepro_aprepro_test_dump_reread&field3=status&compare3=62&value3=passed&field4=status&compare4=62&value4=notrun&field5=groupname&compare5=61&value5=ATDM&field6=buildstarttime&compare6=84&value6=2018-08-30&field7=buildstarttime&compare7=83&value7=2018-07-15), the test `SEACASAprepro_aprepro_test_dump_reread` has not failed in any promoted ATDM Trilinos since 8/2/2018.  Perhaps this is resolved?

**Comment by bartlettroscoe** [comment]
FYI: As shown in [this query](https://testing-vm.sandia.gov/cdash/queryTests.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&date=2018-08-27&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=7&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=testname&compare2=65&value2=SEACAS&field3=status&compare3=62&value3=passed&field4=status&compare4=62&value4=notrun&field5=groupname&compare5=61&value5=ATDM&field6=buildstarttime&compare6=84&value6=2018-08-30&field7=buildstarttime&compare7=83&value7=2018-08-09), the only ATDM group SEACAS tests failing since 8/8/2018 are the tests:

* SEACASExodus_exodus_unit_tests
* SEACASExodus_exodus_unit_tests_nc5_env

mostly on 'white' (except for one failure each on 'chama' and 'serrano' for some reason) and the failures on 'white' are addressed in #3288.

In any case, the test failures called out in this Issue seem to be fixed.  Therefore, I think we can close this issue.

Closing as complete.
